File: dotnet/postpony.api.sdk/Postpony.Api.Sdk/Common/ClientManager.cs
using System;

namespace Postpony.Api.Sdk.Common
{
    public class ClientManager
    {
        private static Func<Client> getCurrent;

        internal static Client Build()
        {
            if (getCurrent == null)
                throw new ArgumentException("未初始化账号信息，请通过调用ClientManager.SetCurrent方法配置账号信息");
            return getCurrent();
        }

        public static void SetCurrent(string apiKey,string apiSecret,string apibaseurl)
        {
            SetCurrent(() => new Client(new ClientConfiguration(apiKey,apiSecret,apibaseurl)));
        }

        public static void SetCurrent(Func<Client> getClient)
        {
            getCurrent = getClient;
        }
    }
}

File: java/postpony-api-sdk/src/main/java/com/postpony/common/ClientManager.java
package com.postpony.common;

public class ClientManager {

    private static Client getCurrent;

    public static Client Build()
    {
        if (getCurrent == null)
        {
            String errMsg="未初始化账号信息，请通过调用ClientManager.SetCurrent方法配置账号信息";
            System.out.println(errMsg);
            throw new IllegalArgumentException(errMsg);
        }


        return getCurrent;
    }

    public static void SetCurrent(String apiKey,String apiSecret,String apibaseurl)
    {
        getCurrent=new Client(new ClientConfiguration(apiKey,apiSecret,apibaseurl));
        SetCurrent(getCurrent);
    }

    public static void SetCurrent(Client getClient)
    {
        getCurrent = getClient;
    }
}
